test/uk/gov/hmrc/softdrinksindustrylevystub/controllers/SubscriptionControllerSpec.scala

```diff
@@ -25,42 +26,344 @@ import play.api.libs.json.Json
+      val input = Json.parse("""{
+                      "organisationType" : "LimitedCompany",
+                      "dateOfApplication" : "2017-09-12",
+                      "taxStartDate" : "2017-09-12",
+                      "customerIdentificationNumber" : "1097172564",
+                      "tradingName" : "some trading name",
+                      "businessContactDetails" : {
+                        "addressNotInUk" : false,
+                        "addressLine1" : "line1",
+                        "addressLine2" : "line2",
+                        "postcode" : "ABC123",
+                        "telephoneNumber" : "123123123",
+                        "emailAddress" : "foo@bar.com"
+                      },
+                      "correspondenceAddressDiffers" : false,
+                      "primaryPerson" : {
+                        "name" : "foo",
+                        "telephoneNumber" : "123123123",
+                        "emailAddress" : "foo@bar.com"
+                      },
+                      "softDrinksIndustryLevyDetails" : {
+                        "activities" : "ContractPacker",
+                        "lessThanMillion" : true,
+                        "smallProducerExemption" : false,
+                        "usesCopacker" : false,
+                        "voluntarilyRegistered" : false
+                      },
+                      "sdilActivity" : {
+                      },
+                      "taxObligationStartDate" : "2017-09-12",
+                      "bankDetails" : {
+                        "directDebit" : false
+                      },
+                      "sites" : [ {
+                        "address" : {
+                          "addressNotInUk" : false,
+                          "addressLine1" : "line1",
+                          "addressLine2" : "line2",
+                          "postcode" : "ABC123",
+                          "telephoneNumber" : "123123123",
+                          "emailAddress" : "foo@bar.com"
+                        }
+                      } ]
+                    }
+                    """)
+
+      when(mockDesSubmissionService.createSubscriptionResponse(any())).thenReturn(CreateSubscriptionResponse("foo", "bar"))
+      val response = mockSubscriptionController.createSubscription()(FakeRequest("POST", "/create-subscription").withBody(input))
 
       status(response) mustBe OK
-      verify(mockDesSubmissionService, times(1)).buildResponse()
-      Json.fromJson[DesSubmissionResult](contentAsJson(response)).getOrElse(DesSubmissionResult(true)) mustBe DesSubmissionResult(false)
+      verify(mockDesSubmissionService, times(1)).createSubscriptionResponse(any())
+      Json.fromJson[CreateSubscriptionResponse](contentAsJson(response)).getOrElse(CreateSubscriptionResponse("bar", "foo")) mustBe CreateSubscriptionResponse("foo", "bar")
+    }
+
+    "return Status: 400 Body: CreateSubscriptionResponse for invalid CreateSubscriptionRequest" in {
```

Comment:
Same as above as whether this body is a CreateSubscriptionResponse or something different?